File: average-salary/averageSalary.go
// https://leetcode.com/problems/average-salary-excluding-the-minimum-and-maximum-salary/

package main

import "fmt"

func main() {
	fmt.Println(average([]int{3000, 2500, 6000, 1000, 8000}))
}

func average(salary []int) float64 {
	min := salary[0]
	max := salary[0]
	totalSalary := 0

	for _, num := range salary {
		if num < min {
			min = num
		}
		if num > max {
			max = num
		}
		totalSalary += num
	}
	averageNum := float64((totalSalary - (min + max))) / float64(len(salary)-2)

	return averageNum

}

File: average-salary/averageSalary.ts
// https://leetcode.com/problems/average-salary-excluding-the-minimum-and-maximum-salary/

function average(salary: number[]): number {
  let min = salary[0];
  let max = salary[0];
  let totalAmount = 0;

  for (let num of salary) {
    if (num < min) {
      min = num;
    }
    if (num > max) {
      max = num;
    }
    totalAmount += num;
  }
  // We have min, max, totalAmount

  const averageNum = (totalAmount - (min + max)) / (salary.length - 2);

  return Number(averageNum.toFixed(5));
}
